pkg/oc/cli/newapp/newapp.go

```diff
@@ -176,6 +178,50 @@ func (p *versionedPrintObj) PrintObj(obj runtime.Object, out io.Writer) error {
 	return printFn(obj)
 }
 
+func nonEmpty(str string, def string) string {
+	if str == "" {
+		return def
+	}
+	return str
+}
+
+func checkResources(discoveryClient discovery.DiscoveryInterface, items []runtime.Object) error {
+	resources, err := discoveryClient.ServerResources()
+	if err != nil {
```

Comment:
@wozniakjan @soltysh  I got here because this particular line didn't tolerate partial discovery. All CLI commands should tolerate partial discovery and only fail if they cannot retrieve sufficient information.  There may be cases where all of discovery must match, but this isn't one of them.